lingon/src/renderer.rs

```diff
@@ -116,36 +57,50 @@ const RECT: [Vertex; 6] = [
     Vertex::new(VPosition::new([-0.5, -0.5])),
 ];
 
+/// Used to wrap the construction of a SpriteSheet.
+///
+/// I have some ideas of typing this harder to make sure
+/// you place a tile dimension on it. #CatchThoseMistakes
+///
+/// The struct might also be useless, and we could easily just
+/// give out a SpriteSheet directly.
 #[derive(Clone, Debug)]
 pub struct SpriteSheetBuilder {
     pub image: Vec<u8>,
-    pub image_dim: (usize, usize),
-    pub tile_dim: Option<(usize, usize)>,
+    pub image_dim: (Pixels, Pixels),
+    pub tile_dim: Option<(Pixels, Pixels)>,
 }
 
+/// A marker type for the unit pixels.
+pub type Pixels = usize;
+
 impl SpriteSheetBuilder {
-    pub fn new(width: usize, height: usize, image: Vec<u8>) -> Self {
+    pub fn new(width: Pixels, height: Pixels, image: Vec<u8>) -> Self {
         Self {
             image,
             image_dim: (width, height),
             tile_dim: None,
         }
     }
 
-    pub fn tile_size(mut self, tile_width: usize, tile_height: usize) -> Self {
+    pub fn tile_size(mut self, tile_width: Pixels, tile_height: Pixels) -> Self {
         self.tile_dim = Some((tile_width, tile_height));
         self
     }
 }
 
+/// A light reference to a SpriteSheet that lives on the GPU,
+/// it's practically free to pass around.
```

Comment:
```suggestion
/// A light reference to a SpriteSheet that lives on the GPU.
```

Implicit because of Copy.